packages/test/test-end-to-end-tests/src/test/gc/gcDatastoreDuplicateRoutes.spec.ts

```diff
@@ -57,4 +61,52 @@ describeNoCompat("GC Data Store Duplicates", (getTestObjectProvider) => {
         assert(gcObject.type === SummaryType.Handle, "Expected a handle!");
         assert(gcObject.handleType === SummaryType.Tree, "Expected a gc tree handle!");
     });
+
+    it("Back routes added by GC are removed when passed from data stores to DDSes", async () => {
+        const settings = {};
+        const configProvider = mockConfigProvider(settings);
+        const newConfig: ITestContainerConfig = {
+            runtimeOptions: defaultGCConfig.runtimeOptions,
+            loaderProps: { configProvider },
+        };
+        settings["Fluid.GarbageCollection.TrackGCState"] = "false";
+        mainContainer = await provider.makeTestContainer(newConfig);
+        mainDataStore = await requestFluidObject<ITestDataObject>(mainContainer, "default");
+        await waitForContainerConnection(mainContainer);
+
+        const dds = SharedMap.create(mainDataStore._runtime);
+        mainDataStore._root.set("dds", dds.handle);
+
+        const summarizer1 = await createSummarizer(provider, mainContainer, undefined, undefined, configProvider);
+        let summaryResult = await waitForSummary(summarizer1);
+
+        // Change ds1 but not the root dds
+        dds.set("change", "change1");
+
+        summarizer1.close();
+        const summarizer2 = await createSummarizer(
+            provider,
+            mainContainer,
+            summaryResult.summaryVersion,
+            undefined,
+            configProvider,
+        );
+
+        // Get GC State
+        summaryResult = await waitForSummary(summarizer2);
+        const gcTree = summaryResult.summaryTree.tree[gcTreeKey];
+        assert(gcTree !== undefined, "Expected a gc blob!");
```

Comment:
This should be expected a gc tree. Infact you can combine these two asserts into one:
assert(gcTree?.type === SummaryType.Tree)